pkg/sdk/pipes.go

```diff
@@ -0,0 +1,190 @@
+package sdk
+
+import (
+	"context"
+	"database/sql"
+	"time"
+)
+
+type Pipes interface {
+	// Create creates a pipe.
+	Create(ctx context.Context, id SchemaObjectIdentifier, copyStatement string, opts *PipeCreateOptions) error
+	// Alter modifies an existing pipe.
+	Alter(ctx context.Context, id SchemaObjectIdentifier, opts *PipeAlterOptions) error
+	// Drop removes a pipe.
+	Drop(ctx context.Context, id SchemaObjectIdentifier) error
+	// Show returns a list of pipes.
+	Show(ctx context.Context, opts *PipeShowOptions) ([]*Pipe, error)
+	// ShowByID returns a pipe by ID.
+	ShowByID(ctx context.Context, id SchemaObjectIdentifier) (*Pipe, error)
+	// Describe returns the details of a pipe.
+	Describe(ctx context.Context, id SchemaObjectIdentifier) (*Pipe, error)
+}
+
+// PipeCreateOptions contains options for creating a new pipe in the system for defining the COPY INTO <table> statement
+// used by Snowpipe to load data from an ingestion queue into tables.
+//
+// Based on https://docs.snowflake.com/en/sql-reference/sql/create-pipe.
+type PipeCreateOptions struct {
+	create      bool                   `ddl:"static" sql:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace   *bool                  `ddl:"keyword" sql:"OR REPLACE"`
+	pipe        bool                   `ddl:"static" sql:"PIPE"` //lint:ignore U1000 This is used in the ddl tag
+	IfNotExists *bool                  `ddl:"keyword" sql:"IF NOT EXISTS"`
+	name        SchemaObjectIdentifier `ddl:"identifier"`
+
+	AutoIngest       *bool   `ddl:"parameter" sql:"AUTO_INGEST"`
+	ErrorIntegration *string `ddl:"parameter,no_quotes" sql:"ERROR_INTEGRATION"`
+	AwsSnsTopic      *string `ddl:"parameter,single_quotes" sql:"AWS_SNS_TOPIC"`
+	Integration      *string `ddl:"parameter,single_quotes" sql:"INTEGRATION"`
+	Comment          *string `ddl:"parameter,single_quotes" sql:"COMMENT"`
+
+	as            bool   `ddl:"static" sql:"AS"` //lint:ignore U1000 This is used in the ddl tag
+	copyStatement string `ddl:"keyword,no_quotes"`
+}
+
+// PipeAlterOptions contains options for modifying a limited set of properties for an existing pipe object.
+//
+// Based on https://docs.snowflake.com/en/sql-reference/sql/alter-pipe.
+type PipeAlterOptions struct {
+	alter    bool                   `ddl:"static" sql:"ALTER"` //lint:ignore U1000 This is used in the ddl tag
+	role     bool                   `ddl:"static" sql:"PIPE"`  //lint:ignore U1000 This is used in the ddl tag
+	IfExists *bool                  `ddl:"keyword" sql:"IF EXISTS"`
+	name     SchemaObjectIdentifier `ddl:"identifier"`
+
+	// One of
+	Set       *PipeSet       `ddl:"list,no_parentheses" sql:"SET"`
+	Unset     *PipeUnset     `ddl:"list,no_parentheses" sql:"UNSET"`
+	SetTags   *PipeSetTags   `ddl:"list,no_parentheses" sql:"SET TAG"`
```

Comment:
Yeah, I know that it is different, but tbh I like this approach much more:
- we are more explicit about the fact that these are two completely different alter statements (you cannot set tags and other params in one query)
- validations become easier, because of the above point
- this follows more the documentation where you have 5 different alter statements: https://docs.snowflake.com/en/sql-reference/sql/alter-pipe#syntax